File: frontend/src/types/crawl.ts
// ---- Crawl Log Entry ----
export interface CrawlLogEntry {
  id: string;
  source: string;
  started_at: string;
  finished_at: string | null;
  items_found: number;
  items_new: number;
  items_updated: number;
  status: 'running' | 'success' | 'partial' | 'failed';
  error_message: string | null;
}

// ---- Crawl Job Status ----
export interface CrawlJobStatus {
  source: string;
  enabled: boolean;
  interval_minutes: number | null;
  cron: string | null;
  last_run: string | null;
  next_run: string | null;
  last_status: string | null;
}

// ---- Crawl Status Response ----
export interface CrawlStatusResponse {
  jobs: CrawlJobStatus[];
}

// ---- Crawl Trigger Response ----
export interface CrawlTriggerResponse {
  source: string;
  message: string;
  triggered_at: string;
}

// ---- Crawl Log Query Params ----
export interface CrawlLogQueryParams {
  page?: number;
  page_size?: number;
  source?: string;
  status?: string;
  sort?: string;
}

File: backend/app/schemas/crawl.py
"""Crawl Pydantic schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CrawlLogEntry(BaseModel):
    id: str
    source: str
    started_at: datetime
    finished_at: Optional[datetime] = None
    items_found: int
    items_new: int
    items_updated: int
    items_filtered: int = 0
    items_summarized: int = 0
    status: str
    error_message: Optional[str] = None

    model_config = {"from_attributes": True}


class CrawlStatusJob(BaseModel):
    source: str
    enabled: bool
    interval_minutes: Optional[int] = None
    cron: Optional[str] = None
    last_run: Optional[datetime] = None
    next_run: Optional[datetime] = None
    last_status: Optional[str] = None


class CrawlStatusResponse(BaseModel):
    jobs: list[CrawlStatusJob]


class CrawlTriggerResponse(BaseModel):
    source: str
    message: str
    triggered_at: datetime
